components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa/src/Publisher/Extensions/OpcNodeModelEx.cs

```diff
@@ -33,28 +33,26 @@ public static class OpcNodeModelEx {
             if (string.Compare(model.Id, that.Id, StringComparison.OrdinalIgnoreCase) != 0) {
                 return false;
             }
+
             if (string.Compare(model.DisplayName, that.DisplayName, StringComparison.OrdinalIgnoreCase) != 0) {
                 return false;
             }
+
             if (string.Compare(model.DataSetFieldId, that.DataSetFieldId, StringComparison.OrdinalIgnoreCase) != 0) {
                 return false;
             }
+
             if (string.Compare(model.ExpandedNodeId, that.ExpandedNodeId, StringComparison.OrdinalIgnoreCase) != 0) {
                 return false;
             }
-            if (defaultPublishing.HasValue) {
-                if (model.OpcPublishingInterval.GetValueOrDefault(defaultPublishing.Value) !=
-                    that.OpcPublishingInterval.GetValueOrDefault(defaultPublishing.Value)) {
-                    return false;
-                }
-            }
-            else {
-                if (model.OpcPublishingInterval != that.OpcPublishingInterval) {
-                    return false;
-                }
+
+            if (model.OpcPublishingIntervalTimespan.GetTimeSpanFromMiliseconds(model.OpcPublishingInterval) !=
```

Comment:
We can have `GetOpcPublishingIntervalTimespan()` extension method on the model itself. That one then would not need to get any additional parameters as it will have access to both `OpcPublishingIntervalTimespan` and `OpcPublishingInterval` fields. And the call would also be simplified then as it will look like this:

```
model.GetOpcPublishingIntervalTimespan();
```

Internally it can use the existing `GetTimeSpanFromMiliseconds()` call itself.

I think that this kind of helper methods would also be useful for sampling and heartbeat intervals, `GetOpcSamplingIntervalTimespan()` and `GetHeartbeatIntervalTimespan()`.

Reply:
+1 to have extension methods

Reply:
done